app/controllers/subsidy/applications/edit.js

```diff
@@ -28,6 +37,95 @@ export default class SubsidyApplicationsEditController extends Controller {
   get canDelete() {
     return this.model.consumption.get('status.isConcept');
   }
+  @action
+  async exportSubsidyAsPDF() {
+    const previousDocumentTitle = document.title;
+    const filename = `${await this.createFilename()}.pdf`;
+
+    document.title = filename;
+
+    window.print();
+    window.addEventListener(
+      'afterprint',
+      (document.title = previousDocumentTitle)
+    );
+  }
+
+  prepareTextareasForPrinting() {
+    // Remove any previously created print divs
+    const existingPrintDivs = document.querySelectorAll(
+      '.textarea, .display-on-print'
+    );
+    existingPrintDivs.forEach((div) => div.remove());
+
+    // Store original textareas
+    const textareas = document.querySelectorAll('textarea');
+
+    // Replace textareas with divs
+    textareas.forEach((textarea) => {
+      const div = document.createElement('div');
+      div.textContent = textarea.value;
+      div.style.whiteSpace = 'pre-wrap';
+      div.style.border = '1px solid #ccc';
+      div.style.padding = '5px';
+      div.style.minHeight = `${textarea.offsetHeight}px`;
+      div.classList.add('display-on-print');
+      div.classList.add('textarea', 'display-on-print');
+
+      textarea.classList.add('au-u-hide-on-print');
+      textarea.parentNode.insertBefore(div, textarea);
+    });
+  }
+
+  @action
+  async collectDownloadLinks() {
+    // Get all attachments based on the data-test-file-card-download attribute
+    let elements = document.querySelectorAll(
+      '[data-test-file-card-download=""]'
+    );
+    this.downloadLinks = Array.from(elements).map((link) => ({
+      url: link.href,
+      filename: link.getAttribute('download'),
+    }));
+  }
+
+  async createFilename() {
+    // Get the subsidy name and selected step name so the zip download can look like '<subsidy name> - <subsidy step name>.zip'
+    const currentStepID = this.router.currentRoute.parent.params.step_id;
+    const currentStep = await this.store.findRecord(
+      'subsidy-application-flow-step',
+      currentStepID,
+      {
+        include: 'subsidy-procedural-step',
+      }
+    );
+    const currentProceduralStep = await currentStep.subsidyProceduralStep;
+    const currentProceduralStepName = currentProceduralStep.description;
+    const currentSubsidyName = this.consumption.get(
```

Comment:
I wouldn't use `.get` anymore. Just await the relationship first, as you're doing in the other code already.

Reply:
I got a warning when I didn't use get

Reply:
You need to await the relationship before accessing properties. Similar to the `currentStep` code above it.